runtime/tests/checker/arrays_dictionaries_test.go

```diff
@@ -753,6 +753,72 @@ func TestCheckInvalidArrayRemoveLastFromConstantSized(t *testing.T) {
 	assert.IsType(t, &sema.NotDeclaredMemberError{}, errs[0])
 }
 
+func TestCheckArrayIndexOf(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+      fun test(): Int? {
+          let x = [1, 2, 3]
+					return x.firstIndex(of:2)
+      }
+    `)
+
+	require.NoError(t, err)
+}
+
+func TestCheckArrayFirstIndexDoesNotExist(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+      fun test(): Int? {
+          let x = [1, 2, 3]
+					return x.firstIndex(of:5)
+      }
+    `)
+
+	require.NoError(t, err)
+}
+
+func TestCheckArrayFirstIndexWrongType(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+      fun test(): Int? {
+          let x = [1, 2, 3]
+					return x.firstIndex(of:"foo")
+      }
+    `)
+	errs := ExpectCheckerErrors(t, err, 1)
+	assert.IsType(t, &sema.TypeMismatchError{}, errs[0])
+}
+
+func TestCheckInvalidResourceFirstIndex(t *testing.T) {
+
+	t.Parallel()
+
+	_, err := ParseAndCheck(t, `
+      resource X {}
+
+      fun test(): Int? {
+          let xs <- [<-create X()]
+					return <-xs.firstIndex(of: "foo")
```

Comment:
The return value is always `Int?`, what matter is the array element type, so maybe fix some of the errors like the invalid move operation for the returned value, and also make the argument a resource.

```suggestion
          return xs.firstIndex(of: <-create X())
```

This change suggestion also requires updating the error assertions below.